packages/cli-doctor/src/tools/healthchecks/ruby.ts

```diff
@@ -1,19 +1,211 @@
+import execa from 'execa';
+import fs from 'fs';
+import path from 'path';
+import chalk from 'chalk';
+
+import {logger, findProjectRoot} from '@react-native-community/cli-tools';
+
 import versionRanges from '../versionRanges';
 import {doesSoftwareNeedToBeFixed} from '../checkInstallation';
 import {HealthCheckInterface} from '../../types';
+import {inline} from './common';
+
+// Exposed for testing only
+export const output = {
+  OK: 'Ok',
+  NO_GEMFILE: 'No Gemfile',
+  NO_RUBY: 'No Ruby',
+  BUNDLE_INVALID_RUBY: 'Bundle invalid Ruby',
+  UNKNOWN: 'Unknown',
+} as const;
+
+// The Change:
+// -----------
+//
+// React Native 0.72 primarily defines the compatible version of Ruby in the
+// project's Gemfile [1]. It does this because it allows for ranges insteado of
+// having to pin to a version of Ruby.
+//
+// In previous versions the .ruby-version file defined the compatible version,
+// and it was derived in the Gemfile [2]:
+//
+// > ruby File.read(File.join(__dir__, '.ruby-version')).strip
+//
+// Why all of the changes with Ruby?
+// ---------------------------------
+//
+// React Native has had to weigh up a couple of valid concerns:
+//
+// - Cocoapods: we don't control the minimum supported version, although that
+//   was defined almost a decade ago [3]. Practically system Ruby on macOS works
+//   for our users.
+//
+// - Apple may drop support for scripting language runtimes in future version of
+//   macOS [4]. Ruby 2.7 is effectively EOL, which means many supporting tools and
+//   developer environments _may_ not support it going forward, and 3.0 is becoming
+//   the default in, for example, places like our CI. Some users may be unable to
```

Comment:
```suggestion
//   the default in, for example, places like our CI. Some users may be unable to
```

That's not entirely true. CircleCI still supports Ruby 2.7.7... we don't know what will happens in April though.